code/modules/reagents/chemistry/reagents/drug_reagents.dm

```diff
@@ -35,41 +35,139 @@
 	id = "nicotine"
 	description = "Slightly reduces stun times. If overdosed it will deal toxin and oxygen damage."
 	reagent_state = LIQUID
-	color = "#60A584" // rgb: 96, 165, 132
-	addiction_threshold = 30
+	color = rgb(0,0,0)
+	overdose_threshold = 30
 
 /datum/reagent/drug/nicotine/on_mob_life(mob/living/M)
 	if(prob(1))
 		var/smoke_message = pick("You feel relaxed.", "You feel calmed.","You feel alert.","You feel rugged.")
 		M << "<span class='notice'>[smoke_message]</span>"
-	M.AdjustStunned(-1, 0)
+	if(prob(50))
+		M.Jitter(5)
+		M.AdjustStunned(-1, 0)
+		M.AdjustParalysis(-1, 0)
+		M.AdjustWeakened(-1, 0)
 	M.adjustStaminaLoss(-0.5*REM, 0)
 	..()
 	. = 1
 
+/datum/reagent/drug/nicotine/overdose_process(mob/living/M)
+	if(prob(75))
+		return
+	var/effect = rand(1,7)
+	switch(volume)
+		if(1 to 49)
+			switch(effect)
+				if(1 to 3)
+					M.visible_message("<span class = 'danger'><b>[M.name]</b> looks nervous!</span>")
+					M.confused += 15
+					M.adjustToxLoss(2)
+					M.Jitter(10)
+					M.emote("twitch")
+				if(4 to 6)
+					M.visible_message("<span class = 'danger'><b>[M.name]</b> is all sweaty!</span>")
+					M.bodytemperature += rand(15,30)
+					M.adjustToxLoss(3)
+				if(7)
+					M.adjustToxLoss(4)
+					M.emote("twitch")
+					M.Jitter(10)
+		if(50 to INFINITY)
+			switch(effect)
+				if(1 to 3)
+					M << "<span class = 'userdanger'>You can't breathe!</span>"
+					M.emote("gasp")
+					M.adjustOxyLoss(15)
+					M.adjustToxLoss(3)
+					M.Stun(1)
```

Comment:
It would all apply at once anyways. This just makes it apply once, instead of running the same code 5+ times when it could be made more efficient by only running the apply code once.